### PR TITLE
Add Matomo module

### PR DESCRIPTION
**Pull request by tevincent**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/swisstransfer/ui/MainActivity.kt [hunk 1]
```diff
@@ -48,6 +49,7 @@ class MainActivity : ComponentActivity() {
                 MainScreen()
             }
         }
+        trackScreen()
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/MainActivity.kt:52]: Shouldn't this be added in `NewTransferActivity` as well ?

File: Core2/Matomo/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+}
+
+dependencies {
+    api(core2.matomo)
+    api(core2.androidx.core)
+    api(core2.appcompat)
```
Comment (KevinBoulongne) [inline Core2/Matomo/build.gradle.kts:43]: These 2 can be removed.

File: Core2/gradle/core2.versions.toml [hunk 3]
```diff
@@ -1,5 +1,11 @@
 [versions]
 sentry-android = "7.15.0"
+matomo = "4.1.4"
+androidx-core = "1.13.1"
+appcompatVersion = "1.7.0"
```
Comment (KevinBoulongne) [inline Core2/gradle/core2.versions.toml:5]: These 2 can be removed.

File: Core2/gradle/core2.versions.toml [hunk 4]
```diff
@@ -1,5 +1,11 @@
...
 
 [libraries]
 sentry-android = { module = "io.sentry:sentry-android", version.ref = "sentry-android" }
+matomo = { module = "com.github.matomo-org:matomo-sdk-android", version.ref = "matomo" }
+androidx-core = { module = "androidx.core:core-ktx", version.ref = "androidx-core" }
+appcompat = { module = "androidx.appcompat:appcompat", version.ref = "appcompatVersion" }
```
Comment (KevinBoulongne) [inline Core2/gradle/core2.versions.toml:11]: These 2 can be removed.

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: Core2/Matomo/src/main/java/com/infomaniak/matomo/Matomo.kt [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+            }
+        }
+    }
+
+    fun Context.trackUserId(userId: Int) {
+        tracker.userId = userId.toString()
+    }
+
+    //region Track screens
+
```
Comment (KevinBoulongne) [inline Core2/Matomo/src/main/java/com/infomaniak/matomo/Matomo.kt:65]:
```suggestion
```

File: Core2/Matomo/src/main/java/com/infomaniak/matomo/Matomo.kt [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    fun Activity.trackScreen() {
+        TrackHelper.track().screen(this).title(this::class.java.simpleName).with(tracker)
+    }
+
+    fun Context.trackScreen(path: String, title: String) {
+        TrackHelper.track().screen(path).title(title).with(tracker)
+    }
+
```
Comment (KevinBoulongne) [inline Core2/Matomo/src/main/java/com/infomaniak/matomo/Matomo.kt:73]:
```suggestion
```

File: Core2/Matomo/src/main/java/com/infomaniak/matomo/Matomo.kt [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    //endregion
+
+    //region Track events
+
```
Comment (KevinBoulongne) [inline Core2/Matomo/src/main/java/com/infomaniak/matomo/Matomo.kt:77]:
```suggestion
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/swisstransfer/ui/MainActivity.kt (hunk 1 above)
Comment (tevincent) [reply app/src/main/java/com/infomaniak/swisstransfer/ui/MainActivity.kt:52]: It was just for a test but ultimately, yes.

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**